src/index.ts

```diff
@@ -35,31 +39,71 @@ export interface ITeacherPage {
   department: string;
   school: ISchoolFromSearch;
   legacyId: number;
+  wouldTakeAgainPercent: number;
 }
 
-const searchSchool = async (query: string): Promise<ISchoolFromSearch[]> => {
-  const response = await client.request(autocompleteSchoolQuery, {query});
+export const searchSchool = async (
+  query: string
+): Promise<ISchoolFromSearch[]> => {
+  try {
+    const response = await client.request(autocompleteSchoolQuery, {query});
 
-  return response.autocomplete.schools.edges.map((edge: { node: ISchoolFromSearch }) => edge.node);
+    return response.autocomplete.schools.edges.map(
+      (edge: { node: ISchoolFromSearch }) => edge.node
+    );
+  } catch (error: unknown) {
+    if (error instanceof Error) {
+      console.error('Error in searchSchool:', error.message);
```

Comment:
here (and other places) we should handle the error or re-throw instead of ignoring it (this lets the caller decide what to do with it instead of the library deciding for them)

Reply:
I was thinking about returning a `Promise.reject` or we can always add back throw error.

Reply:
since this is an `async` function we don't have to use `Promise.reject`, we can just `throw` like normal